gateway-service-api/build.gradle.kts

```diff
@@ -47,13 +47,13 @@ sourceSets {
 }
 
 dependencies {
-  api("io.grpc:grpc-protobuf:1.36.1")
-  api("com.google.api.grpc:proto-google-common-protos:1.18.1")
-  api("io.grpc:grpc-stub:1.36.1")
+  api("io.grpc:grpc-protobuf:1.39.0")
+  api("com.google.api.grpc:proto-google-common-protos:2.3.2")
+  api("io.grpc:grpc-stub:1.39.0")
   api("javax.annotation:javax.annotation-api:1.3.2")
 
   constraints {
-    api("com.google.guava:guava:30.0-jre") {
+    api("com.google.guava:guava:30.1.1-jre") {
```

Comment:
this constraint shouldn't even exist as far as I can tell - we shouldn't be specifying a guava version in our api unless we're using it.

Reply:
I tried removing it, and changed `ImmutableSet.toImmutableSet()` to `Collectors.toUnmodifiableSet()` at some places.  And, the tests failed again

Reply:
that indicates issues - if we're using it, then we should be declaring it, not constraining it. And our api shouldn't have any code, anyway - looks like the _impl_ is the one using it and not declaring it - but instead depending on the api's constraint.